src/libsync/theme.h

```diff
@@ -561,6 +563,7 @@ class OWNCLOUDSYNC_EXPORT Theme : public QObject
 #ifndef TOKEN_AUTH_ONLY
     mutable QHash<QString, QIcon> _iconCache;
 #endif
+    mutable QHash<QString, QPixmap> _pixmapCache;
```

Comment:
Will that create problems if we use the Theme class in a multithreaded environment?

Reply:
there is `QPixmapCache` in Qt
use it directly

Reply:
@FlexW In that case we need to use `mutex` in that method, or, make it non-const, remove `mutable` here, and, still use `mutex`. Not sure if we want to do it, though.

Reply:
@mgallien Any thoughts ^^?

Reply:
`mutable` is the proper solution if you want to have `pixmapForBackground` be `const`
what about using `QPixmapCache` ?

Reply:
My point was just that if you have a `const` method one could think: Oh great I can use it multithreaded because it does not modify the state, but in fact with `mutable` it modifies the state.